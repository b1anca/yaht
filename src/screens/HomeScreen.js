import React from "react";
import Header from "../components/Header";

const HomeScreen = () => {
  return (
    <>
      <Header />
      <div className="relative mx-auto mt-16 grid w-full max-w-container grid-cols-1 px-4 sm:mt-20 sm:px-6 lg:px-8 xl:mt-32">
        <h1 className="text-lime-600 col-start-1 row-start-2 mt-4 max-w-[36rem] text-4xl font-extrabold tracking-tight sm:text-7xl xl:max-w-[43.5rem]">
          Yet Another Habits Tracker
        </h1>
      </div>
    </>
  );
};

export default HomeScreen;
