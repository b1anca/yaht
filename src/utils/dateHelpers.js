export const getDaysInMonth = (month, year) => {
  const date = new Date(year, month, 1);
  const days = [];
  while (date.getMonth() === month) {
    days.push(new Date(date));
    date.setDate(date.getDate() + 1);
  }
  return days;
};

export const formatDate = (date) => {
  const monthNames = [
    "Jan",
    "Feb",
    "Mar",
    "Apr",
    "May",
    "Jun",
    "Jul",
    "Aug",
    "Sep",
    "Oct",
    "Nov",
    "Dec",
  ];
  const day = date.getDate();
  const monthIndex = date.getMonth();
  const year = date.getFullYear();

  return day + " " + monthNames[monthIndex] + " " + year;
};

export const areDatesEqual = (date1, date2) => {
  return (
    date1.getFullYear() === date2.getFullYear() &&
    date1.getMonth() === date2.getMonth() &&
    date1.getDate() === date2.getDate()
  );
};
