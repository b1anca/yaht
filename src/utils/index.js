export const getRangeFromViewportWidth = () => {
  const viewportWidth = window.innerWidth;

  if (viewportWidth <= 390) {
    return 3;
  } else if (viewportWidth <= 500) {
    return 4;
    // sm
  } else if (viewportWidth <= 640) {
    return 6;
    // md
  } else if (viewportWidth <= 768) {
    return 8;
    // lg
  } else if (viewportWidth <= 1024) {
    return 9;
  } else {
    return 10;
  }
};
