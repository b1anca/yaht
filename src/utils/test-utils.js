import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import userEvent from "@testing-library/user-event";
import { setupStore } from "../app/store";

export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    store = setupStore(preloadedState),
    ...renderOptions
  } = {}
) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <BrowserRouter>{children}</BrowserRouter>
      </Provider>
    );
  }
  return {
    store,
    user: userEvent.setup(),
    ...render(ui, { wrapper: Wrapper, ...renderOptions }),
  };
}
