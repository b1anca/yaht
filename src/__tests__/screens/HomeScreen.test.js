import React from "react";
import { renderWithProviders } from "../../utils/test-utils";
import HomeScreen from "../../screens/HomeScreen";

describe("HomeScreen", () => {
  it("should render correctly", () => {
    const { asFragment } = renderWithProviders(<HomeScreen />);

    expect(asFragment()).toMatchSnapshot();
  });
});
