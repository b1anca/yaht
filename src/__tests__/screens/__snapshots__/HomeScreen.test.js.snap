// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen should render correctly 1`] = `
<DocumentFragment>
  <header
    class="top-0 z-40 w-full backdrop-blur transition-colors duration-500 border-b border-slate-900/10 dark:border-slate-100/10 text-sm font-semibold leading-6"
  >
    <div
      class="py-4 px-6"
    >
      <div
        class="relative flex items-center"
      >
        <a
          aria-current="page"
          class="text-base whitespace-nowrap py-2 px-4 justify-center font-extrabold py-2.5 px-4 text-zinc-600/90 dark:text-zinc-100/90 hover:underline pl-0 active"
          href="/"
        >
          YAHT
        </a>
        <div
          class="relative flex items-center ml-auto"
        >
          <a
            class="text-base whitespace-nowrap py-2 px-4 justify-center font-extrabold py-2.5 px-4 text-zinc-600/90 dark:text-zinc-100/90 hover:underline"
            href="/login"
          >
            Sign in
          </a>
          <a
            class="text-base whitespace-nowrap py-2 px-4 justify-center font-extrabold py-2.5 px-4 w-full shadow rounded-lg text-zinc-100/90 bg-green-700 hover:bg-green-700/90"
            href="/register"
          >
            Register
          </a>
        </div>
      </div>
    </div>
  </header>
  <div
    class="relative mx-auto mt-16 grid w-full max-w-container grid-cols-1 px-4 sm:mt-20 sm:px-6 lg:px-8 xl:mt-32"
  >
    <h1
      class="text-green-700/90 col-start-1 row-start-2 mt-4 max-w-[36rem] text-7xl font-extrabold tracking-tight"
    >
      Yet Another Habits Tracker
    </h1>
  </div>
</DocumentFragment>
`;
